### PR TITLE
Opta Module Input Schemas

### PR DESCRIPTION
**Pull request by QuinnNgo**
# Description
This PR adds [`JSONSchema`](https://json-schema.org/)s to the Opta repository. These will be served from our GraphQL Gateway for consumption by our frontend.

## TODO
The following is a list of related items that will be included in future work, but are not implemented in this PR.
- Modify `opta validate` to use a module's JSON Schema when there is one
- Add tests to ensure that any `user_facing` fields are included in a module's JSON schema

# Safety checklist
* [x] This change is backwards compatible and safe to apply by existing users
* [x] This change will NOT lead to data loss
* [x] This change will NOT lead to downtime who already has an env/service setup

## How has this change been tested, beside unit tests?
Since this will initially only be consumed by our GraphQL Gateway, tests will be in https://github.com/run-x/graphql-gateway

### CODE REVIEW

File: config/registry/google/modules/gcp-gke.yaml [hunk 1]
```diff
@@ -53,7 +53,7 @@ inputs:
   - name: gke_channel
     user_facing: true
     validator: str(required=False)
-    description: The GKE K8s [release channel](https://cloud.google.com/kubernetes-engine/docs/concepts/release-channels) to bind the cluster too. Gives you automatic K8s version management for the lcuster and node pools.
+    description: The GKE K8s [release channel](https://cloud.google.com/kubernetes-engine/docs/concepts/release-channels) to bind the cluster too. Gives you automatic K8s version management for the  and node pools.
```
Comment (rabbitfang) [inline config/registry/google/modules/gcp-gke.yaml:56]:
```suggestion
    description: The GKE K8s [release channel](https://cloud.google.com/kubernetes-engine/docs/concepts/release-channels) to bind the cluster too. Gives you automatic K8s version management for the cluster and node pools.
```

Is this a typo?

**Review by rabbitfang — COMMENTED** [review]
How are these generated? This looks like it will be pretty tedious to keep the json scheme in sync with the yamle spec if its all done manually.

**Comment by QuinnNgo** [comment]
> How are these generated? This looks like it will be pretty tedious to keep the json scheme in sync with the yamle spec if its all done manually.

The current state (where both JSON schema and Yamale are in the codebase) is meant to be temporary -- this PR is the first step in a complete migration from Yamale to JSON schema. The eventual vision is that the entire `inputs` array in each module description is deprecated and is completely replaced by its corresponding json schema, and for that schema to be the single source of truth for both validation and docs generation

**Comment by QuinnNgo** [comment]
@rabbitfang Just added a script that takes these descriptions from our existing files and makes sure they're up to date during CI: opta/json_schema.py.

**Review by juandiegopalomino — APPROVED** [review]

File: config/registry/schemas/common-types/cidr-block.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "$id": "https://app.runx.dev/common-types/cidr-block",
```
Comment (NitinAgg) [inline config/registry/schemas/common-types/cidr-block.json:2]: why is this needed?

**Review by NitinAgg — APPROVED** [review]

File: config/registry/schemas/common-types/cidr-block.json (hunk 2 above)
Comment (QuinnNgo) [reply config/registry/schemas/common-types/cidr-block.json:2]: These are used when schemas reference each other
